avocado/utils/network/interfaces.py

```diff
@@ -571,3 +571,26 @@ def restore_slave_cfg_file(self):
                     os.remove(slave_config)
             except Exception as ex:
                 raise NWException(f"Could not restore the config file {ex}")
+
+    def are_packets_lost(self, peer_ip, options=None):
+        """Check packet loss that occurs during ping.
+
+        Function returns True for 0% packet loss and False
+        if packet loss occurs.
+
+        You must have sudo permissions to run this method
+
+        :param peer_ip: Peer IP address (IPv4 or IPv6)
+        :param options: ping command options such as -c, -f. Default is None
+        """
+        cmd = f'ping -I {self.name} {peer_ip}'
+        if options is not None:
+            cmd = f'{cmd} {options} '
+        try:
+            output = run_command(cmd, self.host, sudo=True)
```

Comment:
Why always running as sudo? I thought that at some point, the last argument of this method would be "sudo", so it could be used *only* when needed. If an user would like just to check for packet loss, without advanced flooding options, is there any need for sudo here?

Reply:
@beraldoleal Sorry i miss understood, updated the code again. Where sudo as True will be passed as an argument whenever needed and default it will be False. And Sudo will be given inside run_command()